openapi/generator_test.go

```diff
@@ -734,11 +734,21 @@ func TestSetServers(t *testing.T) {
 	assert.Equal(t, servers, g.API().Servers)
 }
 
-type CustomUnit struct {
+type customUnit struct {
 }
 
-func (c CustomUnit) ParseExample(v string) interface{} {
-	return fmt.Sprintf("%s USD", v)
+func (c customUnit) ParseExample(v string) (interface{}, error) {
+	return fmt.Sprintf("%s USD", v), nil
+}
+
+type customTime time.Time
+
+func (c customTime) ParseExample(v string) (interface{}, error) {
+	t1, err := time.Parse(time.RFC3339, v)
+	if err != nil {
+		return nil, err
+	}
+	return t1.Format("2006-01-02T15:04:05"), nil
```

Comment:
I think, for the sake of the example, the return should be `t1`, and the type should implement a `String()` and/or `MarshalJson` methods. The goal is to return the concrete type, and leave the marshaling to the implementation.

Reply:
Hmm, not sure. If we refactor, then we would need something like this:

```go
type customTime struct {
	time.Time
}

func (c customTime) String() string {
	return c.Format("2006-01-02T15:04:05")
}

func (c customTime) ParseExample(v string) (interface{}, error) {
	t1, err := time.Parse(time.RFC3339, v)
	if err != nil {
		return nil, err
	}
	return customTime{t1}, nil
}
```

but then we can't test is as nicely in `TestGenerator_parseExampleValue` as the returned value is an object, not a string representation formatted in the the format we like. WDYT?

Reply:
Yes, that's what I thought of in term of implem.
The marshaling/string representation of the custom type can be tested by comparing the output spec, and if we want to add a test of the type-specific `ParseExample` method, we just need to ensure that the returned value has the proper type.

Reply:
Refactored now.